fix(global): rename default status color to custom

mcoker opened the pull request:
fixes https://github.com/patternfly/patternfly/issues/5175

Also makes a status variation required for the alert component. Previously the alert with no variant would give you the "default", aka "custom" status styling. Now the alert without a variation is unstyled and you have to add `.pf-m-custom`.

patternfly-build commented:
Preview: https://patternfly-pr-5418.surge.sh

thatblindgeye commented on src/patternfly/components/Alert/alert.scss:
```diff
@@ -119,6 +124,13 @@
+    --pf-c-alert--BorderTopColor: var(--pf-c-alert--m-custom--BorderTopColor);
+    --pf-c-alert__icon--Color: var(--pf-c-alert--m-custom__icon--Color);
+    --pf-c-alert__title--Color: var(--pf-c-alert--m-custom__title--Color);
+    --pf-c-alert--m-inline--BackgroundColor: var(--pf-c-alert--m-inline--m-custom--BackgroundColor);
```
Doesn't look like `--pf-c-alert--m-inline--m-custom--BackgroundColor` is defined anywhere

thatblindgeye commented on src/patternfly/components/Popover/popover-title-text.hbs:
```diff
@@ -4,7 +4,7 @@
       {{#> screen-reader}}
-      {{#if popover--IsDefaultAlert}}
+      {{#if popover--IsCustomAlert}}
         Default
```
Similar to above, would we want this text to be "Custom" instead?

thatblindgeye reviewed (commented):
Looks good! Only other thing is the screenshot for the "custom alert popover" example needs to be updated.

mcoker commented:
@thatblindgeye great catches! The full-page example screenshots are updated prior to a release, usually with the versioned commit that adds the release notes. Though you can run the job and push to this PR if you'd like! The instructions are at https://github.com/patternfly/patternfly/#update-screenshots Note that it may update more screenshots than this one since that job isn't run with each change and there have been a lot of changes since the last run.

srambach commented on src/patternfly/components/NotificationDrawer/examples/NotificationDrawer.md:
Line 127 - Should add '.pf-m-custom' as an acceptable modifier (or just remove the list and simply say always to use it with a state modifier)

srambach reviewed (approved):
:+1 LGTM

patternfly-build commented:
:tada: This PR is included in version 5.0.0-alpha.34 :tada:

The release is available on:
- [GitHub release](https://github.com/patternfly/patternfly/releases/tag/prerelease-v5.0.0-alpha.34)
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly/v/5.0.0-alpha.34)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
